kotest-framework/kotest-framework-api/src/commonMain/kotlin/io/kotest/core/test/TestCaseConfig.kt

```diff
@@ -6,6 +6,28 @@ import io.kotest.core.listeners.TestListener
 import kotlin.time.Duration
 
 typealias EnabledIf = (TestCase) -> Boolean
+typealias EnabledOrCauseIf = (TestCase) -> IsActive
+
+class IsActive private constructor(val active: Boolean, reason: String) {
+   private val builder = StringBuilder(reason)
+   val reason get() = builder.trim().toString()
+
+   companion object {
+      val active = IsActive(true, "")
+      fun inactive(reason: String) = IsActive(false, reason)
+
+      fun fold(isActives: Iterable<IsActive>): IsActive {
+         return isActives.fold(active) { acc, e ->
+            IsActive(acc.active && e.active, acc.reason).also {
+               if (!e.active) {
+                  it.builder.appendLine()
+                  it.builder.append(e.reason)
+               }
+            }
+         }
+      }
+   }
+}
 
 data class TestCaseConfig(
    val enabled: Boolean = true,
```

Comment:
I didn't end up deprecating `enabled` and `enabledIf` since `active.kt` lets you specify reasons why a test were skipped.